addons/mail/models/mail_channel.py

```diff
@@ -976,14 +973,13 @@ def _set_last_seen_message(self, last_message):
 
     def channel_fetched(self):
         """ Broadcast the channel_fetched notification to channel members
-            :param channel_ids : list of channel id that has been fetched by current user
         """
         for channel in self:
-            if not channel.channel_message_ids.ids:
+            if not channel.message_ids.ids:
                 return
             if channel.channel_type != 'chat':
                 return
-            last_message_id = channel.channel_message_ids.ids[0] # zero is the index of the last message
+            last_message_id = channel.message_ids.ids[0] # zero is the index of the last message
```

Comment:
Now that I see it this assumption about [0] being last can be false. Unless they changed it in ORM, order of x2many fields is not guaranteed unless an explicit search/sorted is done just before, due to how records will be cached in inverse relation (simply at the end, not respecting the _order).